frame/treasury/src/lib.rs

```diff
@@ -552,6 +641,118 @@ decl_module! {
 			Self::payout_tip(hash, tip);
 		}
 
+		#[weight = SimpleDispatchInfo::FixedNormal(150_000_000)]
+		fn propose_bounty(
+			origin,
+			curator: <T::Lookup as StaticLookup>::Source,
+			#[compact] value: BalanceOf<T>,
+			description: Vec<u8>,
+		) {
+			let proposer = ensure_signed(origin)?;
+			let curator = T::Lookup::lookup(curator)?;
+
+			ensure!(description.len() <= MAX_SENSIBLE_REASON_LENGTH, Error::<T>::ReasonTooBig);
+
+			let bond = T::BountyDepositBase::get()
+				+ T::BountyDepositPerByte::get() * (description.len() as u32).into();
+			T::Currency::reserve(&proposer, bond)
+				.map_err(|_| Error::<T>::InsufficientProposersBalance)?;
+
+			let index = Self::bounty_count();
+			BountyCount::put(index + 1);
+
+			let bounty = Bounty {
+				proposer, curator, value, bond, description
+			};
+
+			Bounties::<T>::insert(index, &bounty);
+			BountyStatuses::insert(index, BountyStatus::Proposed);
+
+			Self::deposit_event(RawEvent::BountyProposed(index));
+		}
+
+		/// Reject a bounty proposal. The original deposit will be slashed.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - Two DB clear.
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedOperational(100_000_000)]
+		fn reject_bounty(origin, #[compact] bounty_id: BountyIndex) {
+			T::RejectOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
+
+			ensure!(Self::bounty_statuses(bounty_id) == Some(BountyStatus::Proposed), Error::<T>::UnexpectedStatus);
+			let bounty = <Bounties<T>>::take(&bounty_id).ok_or(Error::<T>::InvalidProposalIndex)?;
+
+			BountyStatuses::remove(bounty_id);
+
+			let value = bounty.bond;
+			let imbalance = T::Currency::slash_reserved(&bounty.proposer, value).0;
+			T::ProposalRejection::on_unbalanced(imbalance);
+
+			Self::deposit_event(Event::<T>::BountyRejected(bounty_id, value));
+		}
+
+		/// Approve a bounty proposal. At a later time, the bounty will be funded and become active
+		/// and the original deposit will be returned.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB change.
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedOperational(100_000_000)]
+		fn approve_bounty(origin, #[compact] bounty_id: ProposalIndex) {
+			T::ApproveOrigin::try_origin(origin)
+				.map(|_| ())
+				.or_else(ensure_root)?;
+
+			ensure!(<Bounties<T>>::contains_key(bounty_id), Error::<T>::InvalidProposalIndex);
+			ensure!(Self::bounty_statuses(bounty_id) == Some(BountyStatus::Proposed), Error::<T>::UnexpectedStatus);
+			BountyStatuses::insert(bounty_id, BountyStatus::Approved);
+		}
+
+		#[weight = SimpleDispatchInfo::FixedOperational(100_000_000)]
+		fn award_bounty(origin, #[compact] bounty_id: ProposalIndex, beneficiary: <T::Lookup as StaticLookup>::Source) {
```

Comment:
We need an alternative to `award_bounty` right?

If the bounty is approved, but the beneficiary does not follow through.

Reply:
there is `propose_bounty` which is used by council to approve the bounty.
this is when the work is done, and `curator` payout the bounty